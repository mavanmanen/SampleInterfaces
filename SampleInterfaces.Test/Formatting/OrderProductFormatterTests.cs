using ApprovalTests;
using ApprovalTests.Reporters;
using SampleInterfaces.Formatting;
using SampleInterfaces.Models;

namespace SampleInterfaces.Test.Formatting;

[UseReporter(typeof(DiffReporter))]
public class OrderProductFormatterTests
{
    [Fact]
    public void Format_HappyFlow_Succeeds()
    {
        // Arrange
        var sut = new OrderProductFormatter();
        var input = new[]
        {
            (
                order: new Order(1, DateTime.Parse("2023-08-24T23:18:00"), new[] { 1, 2, 3 }),
                products: new Product[]
                {
                    new(1, "Product 1", 12.30m),
                    new(2, "Product 2", 52.23m),
                    new(3, "Product 3", 42.12m)
                }
            )
        };
        
        // Act
        var result = sut.Format(input);
        
        // Assert
        Approvals.VerifyJson(result);
    }
}