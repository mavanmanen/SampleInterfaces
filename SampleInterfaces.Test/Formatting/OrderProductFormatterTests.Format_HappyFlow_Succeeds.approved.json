[
    {
        "Item1": {
            "Id": 1,
            "TimeStamp": "2023-08-24T23:18:00",
            "ProductIds": [
                1,
                2,
                3
            ]
        },
        "Item2": [
            {
                "Id": 1,
                "Name": "Product 1",
                "Price": 12.30
            },
            {
                "Id": 2,
                "Name": "Product 2",
                "Price": 52.23
            },
            {
                "Id": 3,
                "Name": "Product 3",
                "Price": 42.12
            }
        ]
    }
]